### ISSUE TITLE
Semicolons and Inline Comments

### ISSUE THREAD

**Issue by jakevdp** [ipython/ipython#8002]
I came across a somewhat strange pattern in the current IPython shell & notebook regarding the use of semicolons to suppress output in the presence of inline comments.

``` python
In [1]: 1 + 1
Out[1]: 2

In [2]: 1 + 1;

In [3]: 1 + 1;  # semicolon ignored if followed by comment
Out[3]: 2

In [4]: 1 + 1  # semicolon within comment suppresses output;

```

I would expect prompt ``[3]`` to show no output, and prompt ``[4]`` to show the output.

**Comment by rgbkrk** [comment]
This sounds awfully similar to the first ever issue I reported to ipython, #2913.

**Comment by senyai** [comment]
Similar issue:
```python
In [20]: time? #()
  File "<ipython-input-20-2e1279a91a14>", line 1
    time? #()
        ^
SyntaxError: invalid syntax
```
I expected to see the docs

**Comment by takluyver** [comment]
Fixed by #8505
